File: daily_python/2452_Words_Within_Two_Edits_of_Dictionary.py
import os
import sys
from typing import List

class Solution:
    def twoEditWords(self, queries: List[str], dictionary: List[str]) -> List[str]:
        ans = []
        for query in queries:
            for target in dictionary:
                diff = 0
                for i in range(len(query)):
                    if query[i] != target[i]:
                        diff += 1
                    if diff > 2:
                        break
                if diff <= 2:
                    ans.append(query)
                    break
        return ans

def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            queries = list(lines[0].split('=')[1].strip()[2:-3].split(','))
            dictionary = list(lines[1].split('=')[1].strip()[2:-3].split(','))
            dataset.append((queries, dictionary))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for queries, dictionary in dataset:
        results.append(solution.twoEditWords(queries, dictionary))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2026/2452_Words_Within_Two_Edits_of_Dictionary.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    bool diff(string& s1, string& s2){
        int n = s1.length();
        int d = 0;
        for (int i = 0; i < n; i++){
            d += (s1[i] != s2[i]);
            if (d > 2) return false;
        }
        return true;
    }
    vector<string> twoEditWords(vector<string>& queries, vector<string>& dictionary) {
        int q = queries.size();
        int n = dictionary.size();
        vector<string> ans;
        for (string& query : queries){
            for (int i = 0; i < n; i++){
                if (diff(query, dictionary[i])){
                    ans.push_back(query);
                    break;
                }
            }
        }
        return ans;
    }
};

int main(){
    vector<string> queries = {"word","note","ants","wood"};
    vector<string> dictionary = {"wood","joke","moat"};
    // vector<string> queries = {"yes"};
    // vector<string> dictionary = {"not"};
    Solution S;
    vector<string> ans = S.twoEditWords(queries, dictionary);
    for (string s : ans) cout << s << " ";
    cout << endl;
    return 0;
}